src/org/rascalmpl/parser/gtd/util/DoubleArrayList.java

```diff
@@ -102,4 +106,35 @@ public void ditryClear(){
 	public int size(){
 		return size;
 	}
+
+	public void sort(Comparator<Pair<F,S>> comparator) {
+		java.util.List<Pair<F,S>> elems = new java.util.ArrayList<>(size);
```

Comment:
an alternative is to work on an array with indices:

```java
int[] indices = new int[size];
var firstCompare = Comparators.comparing((a,b) -> Integer.compare(first[a], first[b]));
var secondCompare = Comparators.comparing((a,b) -> Integer.compare(second[a], second[b]));
Arrays.sort(indices, firstCompare.thenComparing(secondCompare));
// now you only have to reshuffle first & second based on the index
// which apparantly is also a less than "fun" problem:
// https://stackoverflow.com/a/7366196/11098
```

I would only do this is fhit is call in the hot path.